#VRML_SIM R2020b utf8
WorldInfo {
  basicTimeStep 128
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.0037899260444160298 0.9052768790153575 0.42480514213082626 3.1254400344974997
  position -0.010099015509915321 0.9455115057217769 -0.772250962099575
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 10 10
}
DEF ROBOT Robot {
  translation 0.117707 0.04 -0.286743
  children [
    InertialUnit {
      translation 0 0.03 0
      rotation 0 1 0 -1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 1 0
            metalness 0
          }
          geometry DEF IMU Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "imu"
    }
    Gyro {
      translation 0 0.05 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.333333 1 0
            metalness 0
          }
          geometry DEF GYRO Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      boundingObject USE GYRO
    }
    GPS {
      translation 0 0.04 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.333333 0 1
            metalness 0
          }
          geometry DEF GPS Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      boundingObject USE GPS
    }
    Lidar {
      translation 0 0.03 0.09
      rotation 0 1 0 3.141592653589793
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            metalness 0
          }
          geometry DEF LIDAR Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      tiltAngle -0.12
      fieldOfView 1.57
      numberOfLayers 10
      maxRange 4
    }
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint DEF WHEEL1 Solid {
        translation 0.06005186559358913 -3.157723486129075e-05 0.0500207526884499
        rotation 0.011355480419242383 -0.014724716484270891 -0.9998271029477577 4.708723537184392
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        boundingObject USE WHEEL
        physics DEF WHEEL_PH Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint DEF WHEEL2 Solid {
        translation -0.06000627491628889 -1.1261465698166471e-05 0.05001037499794944
        rotation 0.006888329611843957 -0.006479269089461786 -0.9999552839938518 4.7078475462419815
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics USE WHEEL_PH
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint DEF WHEEL3 Solid {
        translation 0.06000581021086107 2.7169818866217477e-06 -0.04998451417276848
        rotation 0.011355081800713616 -0.01472505624181835 -0.9998271024712094 4.708722744097527
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics USE WHEEL_PH
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint DEF WHEEL4 Solid {
        translation -0.060051451242279466 1.2015803057747292e-05 -0.04996466295142506
        rotation 0.006888333402824968 -0.006479218054907053 -0.9999552842984172 4.707846319313928
        children [
          USE WHEEL
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics USE WHEEL_PH
      }
    }
  ]
  boundingObject USE BODY
  physics Physics {
  }
  controller "circle_track_controller"
  controllerArgs [
    ""
  ]
  supervisor TRUE
}
DEF CONES Transform {
  scale 0.06 0.06 0.06
  children [
    TrafficCone {
      translation -6.26972 0 4.46565
      name "traffic cone(1)"
      physics DEF CONE_PH Physics {
      }
    }
    TrafficCone {
      translation -3.333 0 2.36817
      name "traffic cone(9)"
      physics DEF CONE_PH Physics {
      }
    }
    TrafficCone {
      translation 3.35465 0 4.2979
      name "traffic cone(10)"
      physics DEF CONE_PH Physics {
      }
    }
    TrafficCone {
      translation -10.4731 -4.44089e-16 12.5632
      name "traffic cone(11)"
      physics DEF CONE_PH Physics {
      }
    }
    TrafficCone {
      translation -1.17581 0 0.255439
      name "traffic cone(16)"
      physics DEF CONE_PH Physics {
      }
    }
    TrafficCone {
      translation -7.57492 0 11.5234
      name "traffic cone(15)"
      physics DEF CONE_PH Physics {
      }
    }
    TrafficCone {
      translation -3.78669 0 9.63622
      name "traffic cone(13)"
      physics DEF CONE_PH Physics {
      }
    }
    TrafficCone {
      translation -0.410606 0 7.53441
      name "traffic cone(14)"
      physics DEF CONE_PH Physics {
      }
    }
    TrafficCone {
      translation -9.82373 0 5.93449
      name "traffic cone(12)"
      physics DEF CONE_PH Physics {
      }
    }
    TrafficCone {
      translation -29.9815 5.55112e-17 -3.50641
      name "traffic cone(3)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation -14.534 5.55112e-17 -14.3585
      name "traffic cone(4)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation -14.6447 -3.88578e-16 -21.9842
      name "traffic cone(5)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation -8.90417 -3.88578e-16 -12.0795
      name "traffic cone(33)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation -1.7718 -8.32667e-16 -4.44292
      name "traffic cone(34)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation -4.35867 -3.88578e-16 -19.7046
      name "traffic cone(35)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation -0.670367 -3.88578e-16 -17.1538
      name "traffic cone(36)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation -8.99027 -3.88578e-16 -20.7649
      name "traffic cone(37)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation -4.89902 -3.88578e-16 -8.8564
      name "traffic cone(38)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation 5.11397 -3.88578e-16 -4.73873
      name "traffic cone(39)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation 2.39008 -3.88578e-16 -13.6389
      name "traffic cone(42)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation 4.18513 -3.88578e-16 -9.36252
      name "traffic cone(43)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation -23.6709 5.55112e-17 -3.63542
      name "traffic cone(7)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation -22.363 5.55112e-17 -6.70901
      name "traffic cone(25)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation -17.8806 5.55112e-17 -11.7611
      name "traffic cone(26)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation -20.1937 5.55112e-17 -9.31417
      name "traffic cone(27)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation -26.213 5.55112e-17 -14.4834
      name "traffic cone(28)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation -19.7023 5.55112e-17 -20.8831
      name "traffic cone(29)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation -29.4046 5.55112e-17 -8.11412
      name "traffic cone(30)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation -23.7336 5.55112e-17 -17.9538
      name "traffic cone(31)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation -28.0375 5.55112e-17 -11.9016
      name "traffic cone(32)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation 4.99502 5.55112e-17 0.304143
      name "traffic cone(2)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation -13.8918 5.55112e-17 13.1271
      name "traffic cone(8)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation -22.0842 5.55112e-17 -0.0919513
      name "traffic cone(6)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation -19.4199 5.55112e-17 3.18091
      name "traffic cone(17)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation -16.9211 5.55112e-17 5.27127
      name "traffic cone(18)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation -18.2173 5.55112e-17 11.8123
      name "traffic cone(19)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation -21.4712 5.55112e-17 9.28383
      name "traffic cone(20)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation -24.2471 5.55112e-17 6.06671
      name "traffic cone(21)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation -26.6594 5.55112e-17 3.03566
      name "traffic cone(22)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation -13.9505 5.55112e-17 6.51506
      name "traffic cone(23)"
      physics USE CONE_PH
    }
    TrafficCone {
      translation -28.3291 5.55112e-17 -0.157396
      name "traffic cone(24)"
      physics USE CONE_PH
    }
  ]
}
